test/namedCurry.js

```diff
@@ -0,0 +1,111 @@
+var R = require('../source');
+var eq = require('./shared/eq');
+
+describe('namedCurry', function() {
+  it('curries a single value', function() {
+    var f = R.namedCurry(['a', 'b', 'c', 'd'], function({a, b, c, d}) {return (a + b * c) / d;}); // f(12, 3, 6, 2) == 15
+    var g = f({ a: 12 });
+    eq(g({ b: 3, c: 6, d: 2 }), 15);
+  });
+
+  it('curries multiple values', function() {
+    var f = R.namedCurry([['a'], ['b'], ['c'], ['d']], function({a, b, c, d}) {return (a + b * c) / d;}); // f(12, 3, 6, 2) == 15
+    var g = f({ a: 12, b: 3 });
+    eq(g({ c: 6, d: 2}), 15);
+    var h = f({ a: 12, b: 3, c: 6 });
+    eq(h({ d: 2 }), 15);
+  });
+
+  it('allows further currying of a curried function', function() {
+    var f = R.namedCurry(['a', 'b', 'c', 'd'], function({a, b, c, d}) {return (a + b * c) / d;}); // f(12, 3, 6, 2) == 15
+    var g = f({ a: 12 });
+    eq(g({ b: 3, c: 6, d: 2 }), 15);
+    var h = g({ b: 3 });
+    eq(h({ c: 6, d: 2 }), 15);
+    eq(g({ b: 3, c: 6})({ d: 2}), 15);
+  });
+
+  it('properly reports the length of the curried function', function() {
```

Comment:
I don't think this is necessary, given that the length should *always* be one, correct?